src/mudlet.pro

```diff
@@ -230,10 +306,114 @@ macx:LIBS += \
+# "shadow builds" so we now explicitly change directory using native shell
+# command before carrying them out - however Windows cmd.exe uses a different
+# command separator to Powershell (and *nix shells) so the git commands may need
+# tweaking in that situation
+
+# Edbee widget needed in all cases
+win32 {
+    # Use a check explicitly based on where the project file is in the sources
+    !exists("$${_PRO_FILE_PWD_}/../3rdparty/edbee-lib/edbee-lib/edbee-lib.pri") {
+        message("git submodule for required edbee-lib editor widget missing from source code, executing 'git submodule update --init' to get it...")
+        # Lets presume that changing the directory here only relates to the
+        # command following it and that the change does not persist, note that
+        # ';' only works for PowerShell - for cmd.exe the nearest equivalent is
+        # '&'
+        system("cd $${_PRO_FILE_PWD_}\.. & git submodule update --init 3rdparty/edbee-lib")
+    }
+
+    # AppVeyor is used to build Windows binaries and it has to include the
+    # lua-code-formatter {a.k.a "lcf" module} - users should instead use
+    # luarocks to get this and one or two other lua modules that Mudlet looks
+    # for on profile loading/startup ..."
+    APPVEYOR_TEST = $$(APPVEYOR)
+    !isEmpty( APPVEYOR_TEST ) {
+        equals(APPVEYOR_TEST, "True") {
+            # We are running in the AppVeyor CI environment
+            !exists("$${_PRO_FILE_PWD_}/../3rdparty/edbee-lib/edbee-lib/edbee-lib.pri") {
+                message("git submodule for lua formatter for Appveyor CI build missing from source code, executing 'git submodule update --init' to get it...")
+                system("cd $${_PRO_FILE_PWD_}\.. & git submodule update --init 3rdparty/lua-code-formatter")
+            }
+        }
+    }
+} else {
+    # Use a check explicitly based on where the project file is in the sources
+    !exists("$${_PRO_FILE_PWD_}/../3rdparty/edbee-lib/edbee-lib/edbee-lib.pri") {
+        message("git submodule for required edbee-lib editor widget missing from source code, executing 'git submodule update --init' to get it...")
+        # Lets presume that changing the directory here only relates to the
+        # command following it and that the change does not persist
+        system("cd $${_PRO_FILE_PWD_}/.. ; git submodule update --init 3rdparty/edbee-lib")
+    }
+}
+
+contains( DEFINES, INCLUDE_UPDATER ) {
+    win32 {
+        # DBLSQD updater - needed for Windows
+        !exists("$${_PRO_FILE_PWD_}/../3rdparty/dblsqd/dblsqd-sdk-qt.pri") {
+            message("git submodule for optional but wanted Dblsqd updater missing from source code, executing 'git submodule update --init' to get it...")
+            system("cd $${_PRO_FILE_PWD_}\.. & git submodule update --init 3rdparty/dblsqd")
+        }
+    } else {
+        # DBLSQD updater - needed for Linux and possibly MacOS but that needs checking
```

Comment:
Yes, used in macos and windows as well

Reply:
Right, __dblsqd__ is used for ALL of Linux/Windows/MacOs (but NOT for anything else, FreeBSD, Cygwin, whatever) and __Sparkle__/__Sparkle-glue__ is ONLY for MacOS...